### PR TITLE
Feat/lesq 974/onboarding page [LESQ-974]

### PR DESCRIPTION
**Pull request by njblore**
## Description

Music year: 2003

- add new page at /onboarding containing school picker
- refactor the school picker to make the homeschool option optional so we can exclude it on the onboarding page
- redirect to 404 page when feature flag is not enabled for user
- uses school picker components from resource pages - once we know how we're going to be using these components we should move them into a shared space and consolidate logic that is currently duplicated

## How to test

The feature flag is only active when you have added your posthog distinct_id to the rule ([see here for more details](https://www.notion.so/oaknationalacademy/I-want-a-sign-up-feature-flag-d755688e81414868908210e126de8e5a?pvs=4))

1. Go to https://deploy-preview-2659--oak-web-application.netlify.thenational.academy/onboarding
2. When the ff is not active you should be redirected to 404
3. When the ff is active you should see the onboarding page with school picker
4. The school picker on the downloads journey should be unaffected, you should still be able to select homeschool as an option and the placeholder and error messaging should reflect that https://deploy-preview-2659--oak-web-application.netlify.thenational.academy/teachers/programmes/biology-secondary-ks4-foundation-aqa/units/eukaryotic-and-prokaryotic-cells/lessons/cells/downloads?preselected=all

## Screenshots

How it should now look:
Screenshot 2024-08-02 at 14 04 43

### CODE REVIEW

**Review by JBR90 — APPROVED** [review]
Works as expected

File: src/components/TeacherViews/Onboarding/Onboarding.view.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        schools={schools}
+        label={"School"}
+        setSelectedSchool={setSelectedSchool}
+        required={true}
+        withHomeschool={false}
+      />
+      <OakPrimaryButton
+        onClick={
+          (event) => void handleSubmit(onFormSubmit)(event) // https://github.com/orgs/react-hook-form/discussions/8622}
+        }
```
Comment (carlmw) [inline src/components/TeacherViews/Onboarding/Onboarding.view.tsx:94]: 🏄🏻 minor UX/keyboard nav improvement. If this was an `onSubmit` on an enclosing form then the user could hit enter to submit immediately after selecting their school from the picker.

```jsx
<OakFlex
  as="form"
  onSubmit={
    (event) => void handleSubmit(onFormSubmit)(event)
  }
>
  ...

  <OakPrimaryButton
    disabled={formState.errors?.school !== undefined || !formState.isValid}
    width="100%"
    type="submit"
  >
    Continue
  </OakPrimaryButton>
</OakFlex>
```

**Review by carlmw — APPROVED** [review]
Looking good!

**Comment by RBrownsett** [comment]
@njblore nice work - will validation come at a later point? e.g. user does not enter a school and tries to click 'Continue'? Currently in live on the downloads page the school required section goes red.  

PR: 

image

live deployed site:

image

**Comment by njblore** [comment]
@RBrownsett good catch! There is no error messaging yet but it should now turn the school label red and disable the continue button when there is invalid input
Screenshot 2024-08-05 at 09 05 40

**Review by RBrownsett — APPROVED** [review]
Approved - full testing details on the Notion ticket

**Pull request merged**